File: wrappers/kotlin/src/commonMain/kotlin/api/KeyCorrectnessProof.kt
package anoncreds.api

import anoncreds_rs.ObjectHandle
import anoncreds_rs.anoncreds_key_correctness_proof_from_json

class KeyCorrectnessProof
internal constructor(
    handle: ObjectHandle
) : AnoncredsObject(handle){

    companion object{
        fun fromJson(json: String): KeyCorrectnessProof {
            return fromJson(
                json,
                ::KeyCorrectnessProof,
                ::anoncreds_key_correctness_proof_from_json
            )
        }
    }
}

File: wrappers/dotnet/lib/Objects/KeyCorrectnessProof.cs
namespace AnonCredsNet.Objects;

public class KeyCorrectnessProof : AnonCredsObject
{
    internal KeyCorrectnessProof(int handle)
        : base(handle) { }

    internal static KeyCorrectnessProof FromJson(string json) =>
        FromJson<KeyCorrectnessProof>(json);
}
